--- guake	2012-06-07 01:16:37.000000000 +0545
+++ guake-indicator	2012-07-14 12:02:02.485365766 +0545
@@ -158,7 +158,10 @@
         """If the gconf var use_trayicon be changed, this method will
         be called and will show/hide the trayicon.
         """
-        self.guake.tray_icon.set_visible(entry.value.get_bool())
+        if hasattr(self.guake.tray_icon, 'set_status'):
+            self.guake.tray_icon.set_status(entry.value.get_bool())
+        else:
+            self.guake.tray_icon.set_visible(entry.value.get_bool())
 
     def ontop_toggled(self, client, connection_id, entry, data):
         """If the gconf var window_ontop be changed, this method will
@@ -531,13 +534,25 @@
 
         # setting global hotkey and showing a pretty notification =)
         guake.globalhotkeys.init()
-
         # trayicon!
-        img = pixmapfile('guake-tray.png')
-        self.tray_icon = gtk.status_icon_new_from_file(img)
-        self.tray_icon.set_tooltip(_('Guake Terminal'))
-        self.tray_icon.connect('popup-menu', self.show_menu)
-        self.tray_icon.connect('activate', self.show_hide)
+        try:
+            import appindicator
+            self.tray_icon = appindicator.Indicator(_("guake-indicator"), _("guake-tray"), appindicator.CATEGORY_OTHER)
+            self.tray_icon.set_icon("guake-tray")
+            self.tray_icon.set_status(appindicator.STATUS_ACTIVE)
+            menu = self.get_widget('tray-menu')
+            show = gtk.MenuItem(_('Show'))
+            show.set_sensitive(True)
+            show.connect('activate', self.show_hide)
+            show.show()
+            menu.prepend(show)
+            self.tray_icon.set_menu(menu)
+        except ImportError:
+            img = pixmapfile('guake-tray.png')
+            self.tray_icon = gtk.status_icon_new_from_file(img)
+            self.tray_icon.set_tooltip(_('Guake Terminal'))
+            self.tray_icon.connect('popup-menu', self.show_menu)
+            self.tray_icon.connect('activate', self.show_hide)
 
         # adding images from a different path.
         ipath = pixmapfile('guake.png')
